src/wizards/newspack/views/settings/additional-brands/brand-upsert.tsx

```diff
@@ -0,0 +1,409 @@
+					selected={ brand.meta._custom_url || 'yes' }
+					options={ [
+						{
+							label: __( 'Homepage', 'newspack-plugin' ),
+							value: 'yes',
+						},
+						{
+							label: __( 'Default', 'newspack-plugin' ),
+							value: 'no',
+						},
+					] }
+					onChange={ ( _custom_url: string ) =>
+						updateBrand( { meta: { ...brand.meta, _custom_url } } )
+					}
+				/>
+				<div className="newspack-brand__base-url-component">
+					<span>{ baseUrl }</span>
+					<TextControl
+						className="newspack-brand__base-url-component__text-control"
+						label={ __( 'Slug', 'newspack-plugin' ) }
+						hideLabelFromVision
+						withMargin={ false }
+						value={ brand.slug || '' }
+						onChange={ ( slug: string ) => updateBrand( { slug } ) }
+					/>
+				</div>
+			</Card>
+
+			{ /* Front Page Settings */ }
+			<Card noBorder>
+				<RadioControl
+					className="newspack-brand__base-url-radio-control"
+					label={ __( 'Show on Front', 'newspack-plugin' ) }
+					selected={ showOnFrontSelect }
+					options={ [
+						{
+							label: __( 'Latest posts', 'newspack-plugin' ),
+							value: 'no',
+						},
+						{
+							label: __( 'A page', 'newspack-plugin' ),
+							value: 'yes',
+						},
+					] }
+					onChange={ updateShowOnFront }
+				/>
+				{ showOnFrontSelect === 'yes' && (
+					<SelectControl
+						label={ __( 'Homepage URL', 'newspack-plugin' ) }
+						value={ brand.meta._show_page_on_front || 0 }
+						options={ [
+							{
+								label: __( 'Select a Page', 'newspack-plugin' ),
+								value: 0,
+								disabled: true,
+							},
+							...publicPages.map( page => ( {
+								label: page.title.rendered,
+								value: Number( page.id ),
+							} ) ),
+						] }
+						onChange={ ( _show_page_on_front: number ) =>
+							updateBrand( {
+								meta: {
+									...brand.meta,
+									_show_page_on_front:
+										Number( _show_page_on_front ),
+								},
+							} )
+						}
+						required
+					/>
+				) }
+			</Card>
+
+			{ /* Menu Settings */ }
+			<SectionHeader
+				title={ __( 'Menus', 'newspack-plugin' ) }
+				description={ __(
+					'Customize the menus for this brand',
+					'newspack-plugin'
+				) }
+			/>
+			{ menuLocations &&
+				Object.keys( menuLocations ).map( location => (
+					<SelectControl
+						key={ location }
+						label={ menuLocations[ location ] }
+						value={ findSelectedMenu( location ) }
+						options={ [
+							{
+								label: __( 'Same as site', 'newspack-plugin' ),
+								value: 0,
+							},
+							...availableMenus,
+						] }
+						onChange={ ( menuId: number ) =>
+							updateMenus( location, menuId )
+						}
+					/>
+				) ) }
+
+			{ /* Action Buttons */ }
+			<div className="newspack-buttons-card">
+				<Button
+					disabled={ ! isBrandValid }
+					variant="primary"
+					onClick={ () => upsertBrand( Number( brandId ), brand ) }
+				>
+					{ __( 'Save', 'newspack-plugin' ) }
+				</Button>
+				<Button variant="secondary" href="#/">
```

Comment:
I expected to reach `#/additional-brands` from here.

Reply:
Nice catch, this has been fixed. Implementation is a little verbose but better in the long run. See 78c3e3c71c53bca27b70aaf4f571912bc3112b39 & cfeac38ef8807db035678c1668a643a9454f9a2e